appserver/payara-appserver-modules/healthcheck-metrics/src/main/java/fish/payara/healthcheck/microprofile/metrics/MicroProfileMetricsCheck.java

```diff
@@ -0,0 +1,251 @@
+ *  and therefore, elected the GPL Version 2 license, then the option applies
+ *  only if the new code is made subject to such option by the copyright
+ *  holder.
+ */
+package fish.payara.healthcheck.microprofile.metrics;
+
+import fish.payara.microprofile.metrics.MetricsService;
+import fish.payara.nucleus.healthcheck.configuration.MicroProfileMetricsChecker;
+import fish.payara.notification.healthcheck.HealthCheckResultEntry;
+import static fish.payara.notification.healthcheck.HealthCheckResultStatus.GOOD;
+import static fish.payara.notification.healthcheck.HealthCheckResultStatus.WARNING;
+import fish.payara.nucleus.healthcheck.HealthCheckResult;
+import fish.payara.nucleus.healthcheck.preliminary.BaseHealthCheck;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import javax.annotation.PostConstruct;
+import org.eclipse.microprofile.metrics.ConcurrentGauge;
+import org.eclipse.microprofile.metrics.Counter;
+import org.eclipse.microprofile.metrics.Gauge;
+import org.eclipse.microprofile.metrics.Histogram;
+import org.eclipse.microprofile.metrics.Meter;
+import org.eclipse.microprofile.metrics.Metric;
+import org.eclipse.microprofile.metrics.MetricID;
+import org.eclipse.microprofile.metrics.MetricRegistry;
+import org.eclipse.microprofile.metrics.MetricUnits;
+import org.eclipse.microprofile.metrics.SimpleTimer;
+import org.eclipse.microprofile.metrics.Timer;
+import org.glassfish.api.StartupRunLevel;
+import org.glassfish.hk2.runlevel.RunLevel;
+import org.glassfish.internal.api.Globals;
+import org.jvnet.hk2.annotations.Service;
+
+@Service(name = "healthcheck-mpmetrics")
+@RunLevel(StartupRunLevel.VAL)
+public class MicroProfileMetricsCheck
+        extends BaseHealthCheck<HealthCheckMicroProfileMetricstExecutionOptions, MicroProfileMetricsChecker> {
+
+    @PostConstruct
+    public void postConstruct() {
+        postConstruct(this, MicroProfileMetricsChecker.class);
+    }
+
+    @Override
+    public HealthCheckMicroProfileMetricstExecutionOptions constructOptions(MicroProfileMetricsChecker checker) {
+        return new HealthCheckMicroProfileMetricstExecutionOptions(Boolean.valueOf(
+                checker.getEnabled()),
+                Long.parseLong(checker.getTime()),
+                asTimeUnit(checker.getUnit()),
+                checker.getMetricsScope(),
+                checker.getMetricApplicationName(),
+                checker.getMetricName());
+    }
+
+    @Override
+    protected String getDescription() {
+        return "healthcheck.description.MPmetrics";
+    }
+
+    @Override
+    protected HealthCheckResult doCheckInternal() {
+        String registryName = options.getMetricsScope();
+        List<String> metricNames = new ArrayList<>(Arrays.asList(options.getMetricName().split(",")));
+        metricNames.removeAll(Arrays.asList("", null));
```

Comment:
This is slow, I'd remove it from the string when removing whitespace to save 2 passes